src/Support/Clock.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace Spatie\Ray\Support;
+
+use DateTimeImmutable;
+
+interface Clock
```

Comment:
Instead of having an interface and two classes, it might be lighter to only have a single `Clock` class, with a static method `fake` on in to let it behave as a fake class.

Reply:
I actually like the approach of having an interface with 2 separate classes. There happens 'a lot' in the fake class and this way it is clearly and nicely separated. I don't think we need to have freeze methods on the main class or I don't really see the benefits of merging all to one class.

Reply:
Let's still merge it. I looked at the fake class, and I think it's small enough to be one single class. I feel like that in this case a pragmatic approach has no real downsides.